### PR TITLE
Disable preloading runtime script/style by default

### PR DESCRIPTION
**Pull request by westonruter**
See https://github.com/ampproject/amp-toolbox-php/issues/17.

cf. https://github.com/ampproject/amp-toolbox-php/pull/261

### CODE REVIEW

File: packages/optimizer/README.md [hunk 1]
```diff
@@ -333,7 +334,11 @@ Hero images are optimized by server-side rendering the `img` element inside the
 
 #### `preloadHeroImage`
 
-Deprecated, use [optimizeHeroImages](#optimizeheroimages) instead. 
+Deprecated, use [optimizeHeroImages](#optimizeheroimages) instead.
+
+#### `preloadEnabled`
```
Comment (westonruter) [inline packages/optimizer/README.md:339]: Maybe `runtimePreloadEnabled` would be better?

File: packages/optimizer/spec/transformers/valid/RewriteAmpUrls/adds_preloads/input.html [hunk 2]
```diff
@@ -1,3 +1,8 @@
+<!--
+{
+  "preloadEnabled": true
```
Comment (westonruter) [inline packages/optimizer/spec/transformers/valid/RewriteAmpUrls/adds_preloads/input.html:3]: This doesn't seem to be getting recognized.

File: packages/optimizer/spec/transformers/valid/RewriteAmpUrls/adds_esm_and_preloads/input.html [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<!--
+{
+  "esmModulesEnabled": true,
+  "preloadEnabled": true
```
Comment (westonruter) [inline packages/optimizer/spec/transformers/valid/RewriteAmpUrls/adds_esm_and_preloads/input.html:4]: This isn't getting recognized. Why?

File: packages/optimizer/lib/transformers/RewriteAmpUrls.js [hunk 4]
```diff
@@ -60,6 +60,7 @@ const {calculateHost} = require('../RuntimeHostHelper');
 class RewriteAmpUrls {
   constructor(config) {
     this.esmModulesEnabled = config.esmModulesEnabled !== false;
+    this.preloadEnabled = config.preloadEnabled === true;
```
Comment (westonruter) [inline packages/optimizer/lib/transformers/RewriteAmpUrls.js:63]: Should this only be honored if the boilerplate was removed?

**Comment by schlessera** [comment]
I would suggest implementing this option in the following way:

**`preloadRuntime`** (`string`):
- `always` => always preload the runtime script
- `never` => never preload the runtime script
- `auto` (default) => preload the runtime script iff the boilerplate is still present

File: packages/optimizer/spec/transformers/valid/RewriteAmpUrls/adds_esm_and_preloads/input.html (hunk 3 above)
Comment (schlessera) [reply packages/optimizer/spec/transformers/valid/RewriteAmpUrls/adds_esm_and_preloads/input.html:4]: This is passed into the transformer's `transform()` method as the second `params` argument and thus needs to be read from there. You can see that `params.esmModulesEnabled` is already being processed in that method.

File: packages/optimizer/lib/transformers/RewriteAmpUrls.js (hunk 4 above)
Comment (sebastianbenz) [reply packages/optimizer/lib/transformers/RewriteAmpUrls.js:63]: Yes, only if boilertplate has been removed. I'm wondering whether we should make it configurable at all. I'd tend to remove the option.

**Comment by westonruter** [comment]
I've removed the option to control whether preload links are added and now made it conditional based on whether the boilerplate is present.

File: packages/optimizer/lib/transformers/RewriteAmpUrls.js (hunk 4 above)
Comment (westonruter) [reply packages/optimizer/lib/transformers/RewriteAmpUrls.js]: Done in 17cd9bd

**Review by sebastianbenz — APPROVED** [review]
Thanks!

**Pull request merged**